lib/portlayer/exec/handle.go

```diff
@@ -283,6 +284,19 @@ func Create(ctx context.Context, sess *session.Session, config *ContainerCreateC
 		Metadata: config.Metadata,
 	}
 
+	dsType, err := sess.Datastore.Type(ctx)
```

Comment:
this is the place to do this: https://github.com/vmware/vic/blob/master/lib/spec/disk.go#L64
and session.go already provide method IsVSAN